### PR TITLE
Comments show up on the page when the youtube.html page is loaded

### PR DESCRIPTION
**Pull request by miahsanchez**
Screenshot 2020-06-04 at 5 04 12 PM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -29,24 +35,37 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Youtube Idea").addSort("timestamp", SortDirection.ASCENDING);
 
-    // TODO:  Make ArrayList more abstract aka make it a List.
-    ArrayList<String> sampleContents = new ArrayList<String>();
-    sampleContents.add("My name is Miah");
-    sampleContents.add("Im from NYC");
-    sampleContents.add("I love coding!");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: private final instance variable for this as we only need one datastore service instance per DataServlet instance

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -29,24 +35,37 @@
...
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> ideas = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
+        String idea = (String) entity.getProperty("idea");
+
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: nit: remove extra blank line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -29,24 +35,37 @@
...
+        ideas.add(idea);
+    }
+
+    Gson gson = new Gson();
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: Lets actually also make this a private final instance variable, we should need only one Gson instance 
Check out the javadoc for the Gson class for why that is: https://www.javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/Gson.html 
(and also we can talk about reading javadocs at tomorrow's standup which is a good topic of discussion!) :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -29,24 +35,37 @@
...
-      response.setContentType("text/html;");
-      response.getWriter().println("Thanks for the idea! If I ever make a video about " + userInput + " you'll be the first to know");
+      String userIdea = request.getParameter("user-input");
+      long timestamp = System.currentTimeMillis();
+
+      Entity youtubeEntity = new Entity("Youtube Idea");
+      youtubeEntity.setProperty("idea", userIdea);
+      youtubeEntity.setProperty("timestamp", timestamp);
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:66]: this line can be removed once you've made this a private final instance variable

File: portfolio/src/main/webapp/youtube.html [hunk 5]
```diff
@@ -4,14 +4,15 @@
...
     <title>Youtube</title>
     <link rel="stylesheet" href="youtube-style.css">
-    <script src="youtube_script.js"></script>
+    <script src="script.js"></script>
   </head>
-  <body>
+  <body onload="loadComments()">
       <form action="/data" method="POST">
         <label for="user-input">What's a youtube video idea you have for me?</label><br>
         <textarea name="user-input">Put your idea here!</textarea>
```
Comment (surbhimadan1995) [inline portfolio/src/main/webapp/youtube.html:12]: make this "required"

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (miahsanchez) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: i did this in my post-practice branch which just got approved so I will merge that now!

File: portfolio/src/main/webapp/youtube.html (hunk 5 above)
Comment (miahsanchez) [reply portfolio/src/main/webapp/youtube.html:12]: i also have this in my post-practice branch! merging that branch now!

**Pull request merged**